src/main/java/com/google/api/generator/gapic/composer/SampleCodeHelperComposer.java

```diff
@@ -0,0 +1,223 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.gapic.composer;
+
+import com.google.api.generator.engine.ast.AssignmentExpr;
+import com.google.api.generator.engine.ast.CommentStatement;
+import com.google.api.generator.engine.ast.Expr;
+import com.google.api.generator.engine.ast.ExprStatement;
+import com.google.api.generator.engine.ast.LineComment;
+import com.google.api.generator.engine.ast.MethodInvocationExpr;
+import com.google.api.generator.engine.ast.Statement;
+import com.google.api.generator.engine.ast.TryCatchStatement;
+import com.google.api.generator.engine.ast.TypeNode;
+import com.google.api.generator.engine.ast.Variable;
+import com.google.api.generator.engine.ast.VariableExpr;
+import com.google.api.generator.gapic.model.Method;
+import com.google.api.generator.gapic.model.MethodArgument;
+import com.google.api.generator.gapic.model.ResourceName;
+import com.google.api.generator.gapic.utils.JavaStyle;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public final class SampleCodeHelperComposer {
+  private static String RESPONSE_VAR_NAME = "response";
+
+  public static TryCatchStatement composeRpcMethodSampleCode(
+      Method method,
+      List<MethodArgument> arguments,
+      TypeNode clientType,
+      Map<String, ResourceName> resourceNames) {
+    // Default Unary RPC method.
+    if (arguments.isEmpty()) {
+      return composeUnaryRpcDefaultMethodSampleCode(method, clientType);
+    }
+    // Paged Unary RPC method.
+    if (method.isPaged()) {
+      return composePagedUnaryRpcMethodSampleCode(method, arguments, clientType);
+    }
+    // Long run operation Unary RPC method.
+    if (method.hasLro()) {
+      return composeLroUnaryRpcMethodSampleCode(method, arguments, clientType);
+    }
+    // Pure Unary RPC method.
+    return composeUnaryRpcMethodSampleCode(method, arguments, clientType, resourceNames);
+  }
+
+  private static TryCatchStatement composeUnaryRpcMethodSampleCode(
+      Method method,
+      List<MethodArgument> arguments,
+      TypeNode clientType,
+      Map<String, ResourceName> resourceNames) {
+    // TODO(summerji): Add unit tests.
+    // Assign each method arguments with default value.
+    List<Statement> bodyStatements =
+        arguments.stream()
+            .map(
+                methodArg ->
+                    ExprStatement.withExpr(
+                        assignMethodArgumentWithDefaultValue(methodArg, resourceNames)))
```

Comment:
Left this comment in subsequent PRs. We should generally reuse the VariableExprs instead of recreating them multiple times. For instance, this could be a list of Exprs instead, which we would then map into AssignmentExprs -> Statements. Then we can pass the list of Exprs into the method invocation.

Same re-creation comment applies throughout.

Reply:
emm, if we use list of Exprs as body, it makes hard for insert a CommentStament, or breakstatement in the middle.
For example, when we need a comment in the middle
```
   * try (AutoscalingPolicyServiceClient autoscalingPolicyServiceClient = AutoscalingPolicyServiceClient.create()) {
   *   RegionName parent = RegionName.of("[PROJECT]", "[REGION]");
   *   ListAutoscalingPoliciesRequest request = ListAutoscalingPoliciesRequest.newBuilder()
   *     .setParent(parent.toString())
   *     .build();
   *   while (true) {
   *     ListAutoscalingPoliciesResponse response = autoscalingPolicyServiceClient.listAutoscalingPoliciesCallable().call(request);
   *     for (AutoscalingPolicy element : response.getPoliciesList()) {
   *       // doThingsWith(element);
   *     }
   *
   *     String nextPageToken = response.getNextPageToken();
   *     if (!Strings.isNullOrEmpty(nextPageToken)) {
   *       request = request.toBuilder().setPageToken(nextPageToken).build();
   *     } else {
   *       break;
   *     }
   *   }
   * }
```
For this situation, in the WhileStatement, the composing order is assignmentExpr, ForStatement, EmptyLineStatement, AssignmentExpr, IfStatement.
How do we handle the mixed situation?

Reply:
Keep a list of statements, and add to it iteratively. Example in pseudocode:

```
VariableExpr responseVarExpr; // Construct and assign value here.
Expr responseValueExpr; // Construct and assign value here.
AssignmentExpr responseAssignExpr = 
    AssignmentExpr.builder().setVariableExpr(responseVarExpr).setValueExpr(responseValueExpr).build();

List<Statement> whileBody = new ArrayList<>();
whileBody.add(ExprStatement.withExpr(responseAssignExpr));

// Construct for loop
whileBody.add(doThingsForStatement);
whileBody.add(EmptyLineStatement.create());

// etc.
```

Reply:
Gotcha, Thanks showing me example.